backend/subletting/models.py

```diff
@@ -0,0 +1,53 @@
+from django.contrib.auth import get_user_model
+from django.db import models
+from django.utils import timezone
+
+
+User = get_user_model()
+
+class Offer(models.Model):
+    user = models.ForeignKey(User, on_delete=models.CASCADE, related_name="offers_made")
+    sublet = models.ForeignKey("Sublet", on_delete=models.CASCADE, related_name="offers")
+    #TODO: check if the way we're handling phone/email is fine, validate on frontend/routes that at least one is not null
+    email = models.EmailField(max_length=255, null=True)
+    phone_number = models.CharField(max_length=20, null=True)
+    message = models.CharField(max_length=255, blank=True)
+    created_date = models.DateTimeField(auto_now_add=True)
+
+    def __str__(self):
+        return f"Offer for {self.sublet} made by {self.user}"
+
+class Amenity(models.Model):
+    name = models.CharField(max_length=100, primary_key=True)
+
+    def __str__(self):
+        return self.name
+
+class Sublet(models.Model):
+    subletter = models.ForeignKey(User, on_delete=models.CASCADE)
+    sublettees = models.ManyToManyField(User, through=Offer, related_name="sublets_offered")
+    amenities = models.ManyToManyField(Amenity)  # TODO: not sure if anything else should go into this as params
+
+    title = models.CharField(max_length=255)
+    address = models.CharField(max_length=255)
+    beds = models.IntegerField()
+    baths = models.IntegerField()
+    description = models.TextField()
```

Comment:
I think we can make this nullable / blank

Reply:
@judtinzhang on a similar vein should we allow saving a "draft", i.e. maybe adding a boolean field like is_draft? although this could also mess with query speed...

Reply:
I like that idea, like having a "Save for later" feature. Perhaps we can do something similar to Polls / Posts (e.g. with an enum)